src/type.js

```diff
@@ -85,6 +88,14 @@ async function typeImpl(
 
   if (!skipClick) click(element)
 
+  if (isContentEditable(element)) {
+    if (document.getSelection().rangeCount === 0) {
+      const range = document.createRange()
+      range.setStart(element, 0)
+      range.setEnd(element, 0)
+      document.getSelection().addRange(range)
+    }
+  }
```

Comment:
Is this not all handled by the `setSelectionRangeIfNecessary` call below?

Reply:
The problem is that when the div has the focus it's possible that `getSelection` is elsewhere. Maybe we can put this logic in the `getSelectionRange`. If the element of `getSelection` is not the current element we can change the range

Reply:
I wouldn't want `getSelectionRange` to perform a side-effect.

How is it possible for the element with focus to not be the one with selection?

Reply:
Good question. I need to create some use cases on playground because there are more thing to consider

Reply:
`jsdom` behaviour is very strange. 

1) jsdom https://repl.it/repls/DutifulLividBytecode#index.js
2) without jsdom https://jsfiddle.net/41sabm23/1/

In `user-event` focus is working right but after it `getSelection().rangeCount` is 0

Reply:
Seems like a jsdom bug you could report (using those exact same reproduction links).

I think we can merge this for now though.